docs/source/sparse.rst

```diff
@@ -167,6 +171,160 @@ receiving a particular layout. We are working on an API to control the result la
 and recognize it is an important feature to plan a more optimal path of execution for
 any given model.
 
+.. _sparse-semi-structured-docs:
+
+Sparse Semi-Structured Tensors
+++++++++++++++++++++++++++++++
+
+.. warning::
+
+   Sparse semi-sturctured tensors are currently a prototype feature and subject to change. Please feel free to open an issue to report a bug or if you have feedback to share. 
+
+Semi-Structured sparsity is a sparsity layout introduced in NVIDIA's Ampere architecutre. 
+This kind of sparsity is also known as **fine-grained structured sparsity** or **2:4 structured sparsity**
+
+In this format, we store `n` out of every `2n` elements, where `n` depends on the width of the dtype of the Tensor. The most commonly used dtype is float16, where `n=2`, hence 2:4 structured sparsity.
+For more information about this layout refer `here <https://developer.nvidia.com/blog/exploiting-ampere-structured-sparsity-with-cusparselt>`_.
+
+PyTorch implements support for semi-structured sparsity via a Tensor subclass.
+We store the sparse tensor in compressed form by keeping only our specified elements as well as some metadata that encodes the mask. The metadata overhead is just 2 bits per element in the tensor. 
```

Comment:
You could add an example calculation here on what that means for the expected compression ratio.